### PR TITLE
add c merge sort content

### PR DESCRIPTION
**Pull request by L-Super**
If this PR is related to coding or code translation, please fill out the checklist.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Review by krahets — COMMENTED** [review]
Thanks! Could you make the code and comments consistent with the reference code (recommended C++)?

**Comment by L-Super** [comment]
> Thanks! Could you make the code and comments consistent with the reference code (recommended C++)?

It's my mistake, I'll check it.

**Comment by L-Super** [comment]
By the way, there are different naming styles in the file, such as  `c/include/print_util.h`.

File: codes/c/chapter_sorting/merge_sort.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    // i和j分别指向两个数组开头部分
+    int i = left, j = mid + 1; 
+    // 通过覆盖原数组 nums 来合并左子数组和右子数组
+    for (int k = left; k <= right; k++) {
+        if (i > mid) {
+            nums[k] = tmp[j - left];
+            j++;
+        }
+        else if (j > right) {
```
Comment (krahets) [inline codes/c/chapter_sorting/merge_sort.c:28]: Still not consistent with C++ code. Please check it.

File: codes/c/chapter_sorting/merge_sort.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * 左子数组区间 [left, mid]
+ * 右子数组区间 [mid + 1, right]
+ */
+void merge(int nums[], int left, int mid, int right) {
+    // 开辟一个新的数组，将原数组数据片段存进去
+    int tmp[right - left + 1]; 
+    for (int m = left; m <= right; m++) {
+        tmp[m - left] = nums[m];
+    }
+    // i和j分别指向两个数组开头部分
```
Comment (krahets) [inline codes/c/chapter_sorting/merge_sort.c:20]: Not consistent

File: codes/c/chapter_sorting/merge_sort.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+#include "../include/include.h"
+
+/**
+ * 合并左子数组和右子数组
+ * 左子数组区间 [left, mid]
+ * 右子数组区间 [mid + 1, right]
+ */
+void merge(int nums[], int left, int mid, int right) {
+    // 开辟一个新的数组，将原数组数据片段存进去
```
Comment (krahets) [inline codes/c/chapter_sorting/merge_sort.c:15]: Not consistent

**Review by krahets — COMMENTED** [review]
Please address the latest comments and check the consistency carefully. It is required to `copy` every comment from C++ to C.

**Comment by krahets** [comment]
> By the way, there are different naming styles in the file, such as `c/include/print_util.h`.

Thanks! I've unified the naming in https://github.com/krahets/hello-algo/commit/40b5be53021d459ad22263b4dcc8e56aff5784fa

**Comment by L-Super** [comment]
Added missing comments, part of the code is inconsistent with C + + , I think it is easier to understand.

**Comment by krahets** [comment]
> Added missing comments, part of the code is inconsistent with C + + , I think it is easier to understand.

It is required to make the code consistent with the C++ codes to avoid some annoying issues caused by the inconsistency. If you want to improve the codes, please create a pull request to clarify the reason and modify the codes of all the languages.

**Comment by krahets** [comment]
Closed due to no response.

**Pull request closed without merging**